frame/vamm/src/tests.rs

```diff
@@ -179,27 +195,99 @@ proptest! {
 			assert_ok!(vamm_created_ok);
 
 			System::assert_last_event(Event::Vamm(
-				pallet::Event::Created { vamm_id: 0u128, state: vamm_created}
+				pallet::Event::Created { vamm_id: 0_u128, state: vamm_created}
 			))
 		});
 	}
 
-
+	#[test]
 	fn create_vamm_updates_storage_map(
-		base_asset_reserves in MINIMUM_RESERVE..=MAXIMUM_RESERVE,
-		quote_asset_reserves in MINIMUM_RESERVE..=MAXIMUM_RESERVE,
-		peg_multiplier in MINIMUM_RESERVE..=MAXIMUM_RESERVE
+		(base_asset_reserves, quote_asset_reserves, peg_multiplier) in min_max_reserve(),
 	) {
 		ExtBuilder::default().build().execute_with(|| {
-			assert!(!VammMap::<MockRuntime>::contains_key(0u128));
+			assert!(!VammMap::<MockRuntime>::contains_key(0_u128));
 
 			let vamm_created_ok = Vamm::create(
 				&VammConfig{base_asset_reserves,
 						   quote_asset_reserves,
 						   peg_multiplier});
 			assert_ok!(vamm_created_ok);
 
-			assert!(VammMap::<MockRuntime>::contains_key(0u128));
+			assert!(VammMap::<MockRuntime>::contains_key(0_u128));
 		});
 	}
 }
+
+// ----------------------------------------------------------------------------------------------------
+//                                             Get Price
+// ----------------------------------------------------------------------------------------------------
+
+proptest! {
+	#![proptest_config(ProptestConfig::with_cases(RUN_CASES))]
+	#[test]
+	#[allow(clippy::disallowed_methods)]
+	fn get_price_base_asset(
+		(base_asset_reserves, quote_asset_reserves, peg_multiplier) in min_max_reserve(),
+	) {
+		ExtBuilder {
+			vamm_count: 1,
+			vamms: vec![
+				(0,
+				 VammState{
+					 base_asset_reserves,
+					 quote_asset_reserves,
+					 peg_multiplier,
+					 closed: None})]
+		}.build().execute_with(|| {
+			let quote_peg = quote_asset_reserves.checked_mul(peg_multiplier);
+			if quote_peg.is_none() {
+				assert_eq!(
+					Vamm::get_price(0, AssetType::Base),
+					Err(DispatchError::Arithmetic(ArithmeticError::Overflow)))
```

Comment:
Does this work instead?
```suggestion
				assert_err!(
					Vamm::get_price(0, AssetType::Base),
					ArithmeticError::Overflow)
```

Reply:
It does.
More concise and clear, thanks for the input!